### PR TITLE
Makes NODROP mops not get stuck in mop buckets

### LINKED ISSUE #26205: Putting a Janitorial Toolset Mop into a Mop Bucket
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1641

### Issue Description

I took my advanced mop and put it into a mop bucket i found in maint. The mop stuck into the mop bucket. Somehow I got it out the first time but the second time around. I pulled the mop out with my left hand and spawned in a new tool with my right hand.

From that point on, i could spawn in new tools but not my mop

### What did you expect to happen?

My mop to not get disconnected from my arm.

### What happened instead?

My mop got permanently disconnected from my arm

### Why is this bad/What are the consequences?

Confusing mop mechanics

### Steps to reproduce the issue.

Find mop bucket, put mop in it. Mop from mop arm is now in mop bucket. Mess around.

### When did the problem start happening?

7/10/2024

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Makes NODROP mops not get put into mop buckets
Also cleans up the entire proc a bit

Also refactors a proc that went
`mopbucket calls proc on mop which calls proc on mopbucket`
Into something sensible

## Why It's Good For The Game
Resolves #26205

## Testing
Put a janitorial implant in me
Tried to put the mop in the bucket, did not work

Checked with a normal mop that it could be inserted

Checked that a cyborg couldn't put their mop in a bucket
image

## Changelog
:cl:
fix: NODROP mops now won't get put in mop buckets
/:cl:

### CODE REVIEW

File: code/game/objects/structures/mop_bucket.dm [hunk 1]
```diff
@@ -23,20 +23,25 @@
...
+/obj/structure/mopbucket/attackby(obj/item/W, mob/user, params)
 	if(W.is_robot_module())
 		to_chat(user, "<span class='warning'>You cannot interface your modules with [src]!</span>")
 		return
 	if(istype(W, /obj/item/mop))
-		var/obj/item/mop/M = W
-		if(M.reagents.total_volume < M.reagents.maximum_volume)
-			M.wet_mop(src, user)
+		var/obj/item/mop/attacking_mop = W
+		if(attacking_mop.reagents.total_volume < M.reagents.maximum_volume)
```
Comment (DGamerL) [inline code/game/objects/structures/mop_bucket.dm:32]:
```suggestion
		if(attacking_mop.reagents.total_volume < attacking_mop.reagents.maximum_volume)
```

File: code/game/objects/structures/mop_bucket.dm [hunk 2]
```diff
@@ -23,20 +23,25 @@
...
-		if(!stored_mop)
-			M.mopbucket_insert(user, src)
+
+		if(!user.unEquip(attacking_mop))
+			to_chat(user, "<span class='notice'>[attacking_mop] is stuck to your hand!</span>")
 			return
-		to_chat(user, "<span class='notice'>Theres already a mop in the mopbucket.</span>")
-		return
+
+		if(!stored_mop && user.unEquip(attacking_mop))
```
[the remark this answers is not in the document]
Comment (DGamerL) [reply code/game/objects/structures/mop_bucket.dm:40]: Forgor to remove that one

[the remark this answers is not in the document]
Comment (DGamerL) [reply code/game/objects/structures/mop_bucket.dm]: Removed!

**Comment by DGamerL** [comment]
\>Why didn't you notice that this doesn't compile

image

**Review by Sirryan2002 — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Pull request merged**